modules/optimization/test/tests/optimizationreporter/base/tests

```diff
@@ -47,7 +47,178 @@
       cli_args = 'OptimizationReporter/upper_bounds=\'0;0;0 0\''
       max_threads = 1
       expect_err = 'When upper_bounds are given in input file, there must either be a single value per parameter group or a value for every parameter in the group\.'
-      detail = 'parameters and error if an incorrect number of parameters are provided for a group.'
+      detail = ' and error if an incorrect number of parameters are provided for a group; '
+    []
+  []
+  [paramMeshReporter]
+    requirement = "The system shall correctly read in parameter data defined on the mesh "
```

Comment:
Usually if the requirement is vague like this, we should probably separate into different blocks. I suggest one block for constant definition, another for variable definition, another for multiple parameters, and finally one for the errors.

Reply:
I split them up into three groups: one parameter group, two parameter groups and errors.  I also moved the base parameterMesh tests to their own directory.